compaction/task_manager_module.cc

```diff
@@ -128,4 +128,44 @@ future<> shard_upgrade_sstables_compaction_task_impl::run() {
     });
 }
 
+future<> scrub_sstables_compaction_task_impl::run() {
+    _stats = co_await _db.map_reduce0([&] (replica::database& db) -> future<sstables::compaction_stats> {
+        sstables::compaction_stats stats;
+        tasks::task_info parent_info{_status.id, _status.shard};
+        auto& compaction_module = db.get_compaction_manager().get_task_manager_module();
+        auto task = co_await compaction_module.make_and_start_task<shard_scrub_sstables_compaction_task_impl>(parent_info, _status.keyspace, _status.id, db, _column_families, _opts, stats);
+        co_await task->done();
+        co_return stats;
+    }, sstables::compaction_stats{}, std::plus<sstables::compaction_stats>());
+}
+
+tasks::is_internal shard_scrub_sstables_compaction_task_impl::is_internal() const noexcept {
+    return tasks::is_internal::yes;
+}
+
+future<> shard_scrub_sstables_compaction_task_impl::run() {
+    _stats = co_await map_reduce(_column_families, [&] (sstring cfname) -> future<sstables::compaction_stats> {
+        sstables::compaction_stats stats{};
+        tasks::task_info parent_info{_status.id, _status.shard};
+        auto& compaction_module = _db.get_compaction_manager().get_task_manager_module();
+        auto task = co_await compaction_module.make_and_start_task<table_scrub_sstables_compaction_task_impl>(parent_info, _status.keyspace, cfname, _status.id, _db, _opts, stats);
+        co_await task->done();
+        co_return stats;
+    }, sstables::compaction_stats{}, std::plus<sstables::compaction_stats>());
+}
+
+tasks::is_internal table_scrub_sstables_compaction_task_impl::is_internal() const noexcept {
+    return tasks::is_internal::yes;
+}
+
+future<> table_scrub_sstables_compaction_task_impl::run() {
+    auto& cm = _db.get_compaction_manager();
+    auto& cf = _db.find_column_family(_status.keyspace, _status.table);
+    co_await cf.parallel_foreach_table_state(coroutine::lambda([&] (compaction::table_state& ts) mutable -> future<> {
```

Comment:
here too, I don't see why a coroutine::lambda is needed.
it has no capture by values and it's being use in a coroutine frame anyway